core/new-gui/src/app/dashboard/component/top-bar/user-icon/user-login/ngbdmodal-user-login.component.ts

```diff
@@ -99,6 +106,51 @@ export class NgbdModalUserLoginComponent implements OnInit {
       }, () => this.registerErrorMessage = 'Registration failed. Could due to duplicate username.');
   }
 
+  /**
+   * this method will init A Google Oauth instance and gapi
+   */
+  public async initGoogleOauth(): Promise<void> {
```

Comment:
is it possible to use `Observable` instead of `Promise`?

Reply:
It seems both `grantOfflineAccess` and `signIn` provided by GoogleAuth library use Promise. So, it might not be possible to use Observable? Not too familiar with their conversion and Angular though...